src/ds/ds6/ds6-safety.cpp

```diff
@@ -0,0 +1,187 @@
+
+        raw_safety_ep->register_metadata(RS2_FRAME_METADATA_SAFETY_LEVEL2_ORIGIN,
+            make_attribute_parser(&md_safety_info::level2_frame_counter_origin, 
+                md_safety_info_attributes::level2_frame_counter_origin_attribute, md_prop_offset));
+
+        raw_safety_ep->register_metadata(RS2_FRAME_METADATA_SAFETY_LEVEL1_VERDICT,
+            make_attribute_parser(&md_safety_info::level1_verdict, 
+                md_safety_info_attributes::level1_verdict_attribute, md_prop_offset));
+
+        raw_safety_ep->register_metadata(RS2_FRAME_METADATA_SAFETY_LEVEL2_VERDICT,
+            make_attribute_parser(&md_safety_info::level2_verdict, 
+                md_safety_info_attributes::level2_verdict_attribute, md_prop_offset));
+
+        raw_safety_ep->register_metadata(RS2_FRAME_METADATA_SAFETY_HUMAN_VOTE_RESULT,
+            make_attribute_parser(&md_safety_info::human_safety_vote_result, 
+                md_safety_info_attributes::human_safety_vote_result_attribute, md_prop_offset));
+
+        raw_safety_ep->register_metadata(RS2_FRAME_METADATA_SAFETY_HARA_EVENTS,
+            make_attribute_parser(&md_safety_info::hara_events, 
+                md_safety_info_attributes::hara_events_attribute, md_prop_offset));
+
+        raw_safety_ep->register_metadata(RS2_FRAME_METADATA_SAFETY_SOC_FUSA_EVENTS,
+            make_attribute_parser(&md_safety_info::soc_fusa_events, 
+                md_safety_info_attributes::soc_fusa_events_attribute, md_prop_offset));
+
+        raw_safety_ep->register_metadata(RS2_FRAME_METADATA_SAFETY_SOC_FUSA_ACTION,
+            make_attribute_parser(&md_safety_info::soc_fusa_action, 
+                md_safety_info_attributes::soc_fusa_action_attribute, md_prop_offset));
+
+        raw_safety_ep->register_metadata(RS2_FRAME_METADATA_SAFETY_FUSA_EVENT,
+            make_attribute_parser(&md_safety_info::mb_fusa_event, 
+                md_safety_info_attributes::mb_fusa_event_attribute, md_prop_offset));
+
+        raw_safety_ep->register_metadata(RS2_FRAME_METADATA_SAFETY_FUSA_ACTION,
+            make_attribute_parser(&md_safety_info::mb_fusa_action, 
+                md_safety_info_attributes::mb_fusa_action_attribute, md_prop_offset));
+
+        raw_safety_ep->register_metadata(RS2_FRAME_METADATA_SAFETY_CRC32,
+            make_attribute_parser(&md_safety_info::crc32,
+                md_safety_info_attributes::crc32_attribute, md_prop_offset,
+                [](const rs2_metadata_type& param) {
+                    // compute crc and check validity
+                    // TODO - REMI - check next line works well during integration - alignement of struct md_safety_info to 4 is needed
+                    uint8_t* start_of_md_struct = reinterpret_cast<uint8_t*>(&const_cast<rs2_metadata_type&>(param)) - sizeof(md_safety_info) + sizeof(param);
+                    auto computed_crc32 = calc_crc32(start_of_md_struct, sizeof(md_safety_info) - sizeof(md_safety_info::crc32));
```

Comment:
As discussed, we will need to see what we do with this CRC, not sure the user needs it.
IMO it should be checked internally, lets add a TODO and continue